draft-ietf-tls-esni.md

```diff
@@ -464,21 +464,11 @@ accepted by the server.
    } AlertDescription;
 ~~~~
 
-Finally, requirements in {{client-behavior}} and {{server-behavior}} require
-implementations to track, alongside each PSK established by a previous
-connection, whether the connection negotiated this extension with the
-"esni_accept" response type. If so, this is referred to as an "ESNI PSK".
-Otherwise, it is a "non-ESNI PSK". This may be implemented by adding a new field
-to client and server session states.
-
-
 ## Client Behavior {#client-behavior}
 
-### Sending an encrypted SNI {#send-esni}
-
-In order to send an encrypted SNI, the client MUST first select one of
-the server ESNIKeyShareEntry values and generate an (EC)DHE share in the
-matching group. This share will then be sent to the server in the
+In order to initiate a TLS handshake using encrypted SNI, the client MUST first
+select one of the server ESNIKeyShareEntry values and generate an (EC)DHE share
+in the matching group. This share will then be sent to the server in the
 "encrypted_sni" extension and used to derive the SNI encryption key. It does not affect the
 (EC)DHE shared secret used in the TLS key schedule. It MUST also select
 an appropriate cipher suite from the list of suites offered by the
```

Comment:
What is "it" in this context? The previous "It" referred to "this share". I assume that what we mean here is "The client MUST also select"

Reply:
To be fixed in upstream in https://github.com/tlswg/draft-ietf-tls-esni/pull/181. Proposed fix has been merged to this branch in 537e3a1.